common/djangoapps/student/helpers.py

```diff
@@ -814,3 +819,101 @@ def user_has_passing_grade_in_course(enrollment):
     except AttributeError:
         pass
     return False
+
+
+def get_instructors(course_run, marketing_root_url):
+    """
+    Get course instructors.
+    """
+    instructors = []
+    staff = course_run.get('staff', [])
+    for instructor in staff:
+        instructor = {
+            'name': f"{instructor.get('given_name')} {instructor.get('family_name')}",
+            'profile_image_url': instructor.get('profile_image_url'),
+            'organization_name': (instructor.get('position').get('organization_name')
+                                  if instructor.get('position') else ''),
+            'bio_url': f"{marketing_root_url}/bio/{instructor.get('slug')}"
+        }
+        instructors.append(instructor)
+
+    return instructors
+
+
+def _prepare_date_block(block, block_date, user_timezone):
+    """
+    Prepare date block which include assignment related data for this date
+    """
+    timezone_offset = get_time_zone_offset(user_timezone, block_date)
+    block = {
+        'title': block.get('title', ''),
+        'assignment_type': block.get('assignment_type', ''),
+        'assignment_count': 0,
+        'link': block.get('link', ''),
+        'date': block_date,
+        'due_date': block_date.strftime("%a, %b %d, %Y"),
+        'due_time': (f'{block_date.strftime("%H:%M %p")} GMT{timezone_offset}' if block.get('assignment_type') else '')
+    }
+    return block
+
+
+def get_course_dates_for_email(user, course_id, request):
+    """
+    Getting nearest dates from today one would be before today and one
+    would be after today.
+    """
+    user_timezone_locale = user_timezone_locale_prefs(request)
+    user_timezone = timezone(user_timezone_locale['user_timezone'] or str(UTC))
+
+    course = get_course_with_access(user, 'load', course_id)
+    date_blocks = get_course_date_blocks(course, user, request, include_access=True, include_past_dates=True)
+    date_blocks = [block for block in date_blocks if not isinstance(block, TodaysDate)]
+    blocks = DateSummarySerializer(date_blocks, many=True).data
+
+    today = datetime.now(user_timezone)
+    course_date = {
+        'title': '',
+        'assignment_type': '',
+        'link': '',
+        'assignment_count': 0,
+        'date': '',
+        'due_date': today.strftime("%a, %b %d, %Y"),
+        'due_time': ''
+    }
+    course_date_list = [
+        {
+            **course_date,
+        },
+        {
+            **course_date,
+            'date': today,
+        },
+        {
+            **course_date,
+        }
+    ]
```

Comment:
nit: move this to a single line
```suggestion
    course_date_list = [{ **course_date, }, { **course_date, 'date': today }, { **course_date }]
```